ignition-server/controllers/machineconfigserver_ignitionprovider.go

```diff
@@ -478,3 +527,25 @@ func compress(content []byte) ([]byte, error) {
 	}
 	return b.Bytes(), nil
 }
+
+func generateCert(mcsService *corev1.Service) error {
```

Comment:
you should not need to generate certs here:
Certs are generated apriori in the pki function

Reply:
You can use a wildcard to cover all cases AKA
`*.machine-config-server.NAMESPACE.svc.cluster.local`

Reply:
You will need to ensure that fomat is added here though:
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/pki/mcs.go#L13